### ISSUE TITLE
[Question] About log of policy_gradient_loss

### ISSUE THREAD

**Issue by d505** [DLR-RM/stable-baselines3#1943]
### ❓ Question
Hi!

I have a question about ppo's policy_gradient_loss log.
The following part
https://github.com/DLR-RM/stable-baselines3/blob/master/stable_baselines3/ppo/ppo.py#L229-L231

Am I correct in understanding that policy_gradient_loss generally gets smaller as we learn? (It is a loss function and is negative, so in the direction away from 0)
I thought the sum of the advantage would be greater because gradually only good actions would be selected. Therefore, I am not sure which value it will converge to, but I expect it to go more negative than 0.

I couldn't figure it out from the following questions.
https://github.com/DLR-RM/stable-baselines3/issues/602

The reason for my question is that I was doing an experiment and it was converging to 0.
![image](https://github.com/DLR-RM/stable-baselines3/assets/58736165/7c002977-85e3-4d77-ae90-650f424f4c47)

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [x] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [x] If code there is, it is [minimal and working](https://github.com/DLR-RM/stable-baselines3/issues/982#issuecomment-1197044014)
- [x] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by giangbang** [comment]
It should always be around zero, because the code intentionally normalizes the advantage to have zero mean 
https://github.com/DLR-RM/stable-baselines3/blob/master/stable_baselines3/ppo/ppo.py#L223 
Additionally, I think it is bad to interpret `policy_gradient_loss` value as a "loss", policy_gradient_loss is just a convenient way to calculate the policy gradient by utilizing the automatic differentiation of pytorch, which is derived from the theory. This generally has little to do as a "performance metric", to see how the algorithm is learning, look at evaluated rewards.

**Comment by d505** [comment]
Thanks for your reply @giangbang !

I understand that it is near zero.
I think you understand that task performance evaluation and policy_gradient_loss value are not the same. However, I also thought that if policy_gradient_loss did not converge to some low value, then learning was not being done. Is that correct?

**Comment by giangbang** [comment]
As I said, `policy_gradient_loss` is just a convenient way to calculate the gradient of the policy, so its values does not necessarily reflect progress in the learning. 
What does it mean by "a convenient way"?. To update neural networks, we will need gradient, but what is the gradient in reinforcement learning? Unlike other learning settings, RL does not have a "loss function"; it optimizes the cumulative rewards directly instead. The question is how to calculate the gradient of cumulative rewards (this might be harder than it sounds; you simulate a bunch of games, collect the rewards in all those games, then how to calculate the gradient from these rewards, they are just scalars and not differentiable anyway); then there is a theorem to do that called policy gradient theorem.
Looking at the formulation of policy gradient theorem, you will see something like this
![image](https://github.com/DLR-RM/stable-baselines3/assets/44130635/f31736b4-af59-42a1-a735-2e46f7068a5b)
That is, the gradient that we are looking for equals the average of Q values multiplied by the score function of the policy (just a fancy way to call grad of log). At this point, it should be obvious at how we implement this formulation in pytorch with something like this:
```python
Q = estimate_q_val()
loss = log(policy) * Q
loss.backward()
```
The actual implementation use advantage instead of Q value, advantage = Q - average(Q), that's why you see the advantage equal zero. This difference does not change the expected gradient, _but it reduces the gradient variance_.
But does this loss reflects anything? No. It is just a way to calculate gradient using an automatic differentiation library.

**Comment by d505** [comment]
The objective function would be as follows. 
![image](https://github.com/DLR-RM/stable-baselines3/assets/58736165/21eb4e78-ec31-4d6c-8fe7-8a2643fb8d89)
I thought the following equation was the gradient after the transformation of this equation.
![image](https://github.com/DLR-RM/stable-baselines3/assets/58736165/04d60410-a4ef-408b-8024-4aaa3b1e88a1)
Therefore, I thought it would be indicative of the same maximization of the objective function of a normal machine learning algorithm.

**Comment by giangbang** [comment]
@d505 It is more subtle than you think. The things here is that two functions can have the same gradient, but their values are not necessarily the same. 
In any case, it seems this discussion is straying from stable-baselines3.

**Comment by d505** [comment]
Thank you very much. I will think about it.
